### ISSUE TITLE
Pie Chart not working?

### ISSUE THREAD

**Issue by groovecoder** [mozilla/redash#240]
### Issue Summary

Pie chart is broken

### Steps to Reproduce

1. Go to https://sql.telemetry.mozilla.org/queries/17837/source#73402

#### Expected results

Lovely pie chart

#### Actual results

Blank chart space and errors in the console

### Technical details:

* Browser/OS: Firefox 57.0a1 (2017-08-31) (64-bit) on Mac OS

```
Error: t.options.seriesOptions[(intermediate value)] is undefined
s/</<@https://sql.telemetry.mozilla.org/app.f27f6f3e7596ec785933.js:1:172920
s/<@https://sql.telemetry.mozilla.org/app.f27f6f3e7596ec785933.js:1:172842
w.forEach@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:1:211006
s@https://sql.telemetry.mozilla.org/app.f27f6f3e7596ec785933.js:1:172622
$digest@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:72899
$apply@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:74399
l@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:48692
b@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:50822
Ft/</k.onload@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:51294
EventHandlerNonNull*Ft/<@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:51131
p@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:49419
s@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:47183
a@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:66317
s/<@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:66489
$eval@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:74114
$digest@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:72596
$apply@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:74399
EventHandlerNonNull*Ft/<@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:51131
p@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:49419
s@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:47183
a@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:66317
s/<@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:66489
$eval@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:74114
$digest@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:72596
$apply@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:74399
l@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:48692
EventHandlerNonNull*Ft/<@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:51131
p@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:49419
s@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:47183
a@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:66317
s/<@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:66489
$eval@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:74114
$digest@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:72596
$apply@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:74399
o/<@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:8141
a@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:18645
o@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:8062
re@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:8351
ie@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:7595
@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:167904
n@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:100441
Ve@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:15330
n@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:15279
EventListener.handleEvent*hr@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:99564
c@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:103023
on@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:103092
Ce.prototype[t]@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:104899
ready@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:100507
@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:167872
@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:64:2
n@https://sql.telemetry.mozilla.org/manifest.763690799c03f663eb20.js:1:96
@https://sql.telemetry.mozilla.org/vendor.c70657ad86bbc02e8dfd.js:24:71583
n@https://sql.telemetry.mozilla.org/manifest.763690799c03f663eb20.js:1:96
@https://sql.telemetry.mozilla.org/app.f27f6f3e7596ec785933.js:1:67791
n@https://sql.telemetry.mozilla.org/manifest.763690799c03f663eb20.js:1:96
window.webpackJsonp@https://sql.telemetry.mozilla.org/manifest.763690799c03f663eb20.js:1:418
@https://sql.telemetry.mozilla.org/app.f27f6f3e7596ec785933.js:1:1
angular.js:13920
o/<
angular.js:13920
bt/this.$get</<
angular.js:10467
$digest
angular.js:17542
$apply
angular.js:17790
l
angular.js:11831
b
angular.js:12033
Ft/</k.onload
angular.js:11966
(Async: EventHandlerNonNull) Ft/<
angular.js:11949
p
angular.js:11776
s
angular.js:11571
a
angular.js:16383
s/<
angular.js:16399:25
$eval
angular.js:17682
$digest
angular.js:17495
$apply
angular.js:17790
(Async: EventHandlerNonNull) Ft/<
angular.js:11949
p
angular.js:11776
s
angular.js:11571
a
angular.js:16383
s/<
angular.js:16399:25
$eval
angular.js:17682
$digest
angular.js:17495
$apply
angular.js:17790
l
angular.js:11831
(Async: EventHandlerNonNull) Ft/<
angular.js:11949
p
angular.js:11776
s
angular.js:11571
a
angular.js:16383
s/<
angular.js:16399:25
$eval
angular.js:17682
$digest
angular.js:17495
$apply
angular.js:17790
o/<
angular.js:1761
a
angular.js:4718
o
angular.js:1759
re
angular.js:1779
ie
angular.js:1664
<anonymous>
angular.js:31763
n
angular.js:3207
Ve
angular.js:3497
n
angular.js:3485
(Async: EventListener.handleEvent) hr
angular.js:2796
c
angular.js:3547
on
angular.js:3560
Ce.prototype[t]
angular.js:3728
ready
angular.js:3214
<anonymous>
angular.js:31762
<anonymous>
angular.js:6
n
bootstrap%2093aa0b5bbd33c1f9e286:54
<anonymous>
index.js:1
n
bootstrap%2093aa0b5bbd33c1f9e286:54
<anonymous>
index.js:14
n
bootstrap%2093aa0b5bbd33c1f9e286:54
window.webpackJsonp
bootstrap%2093aa0b5bbd33c1f9e286:25
<anonymous>
```

**Comment by alison985** [comment]
@groovecoder  I have this fixed on my local with PR #241 but there is a part of production's behavior I haven't been able to reproduce. We're currently pushing m9.1 to production to deal with the autocomplete performance issues. Once that is done, I will push this to staging for further testing.

**Comment by madalincm** [comment]
Verified in FF57(Win7) in STMO-stage
I have created some test queries verify this issue and I have the following scenarios:

1. For some queries the pie chart is nod displayed depending on the numbers of results
example 1:
35 results - the pie chart is displayed: https://pipeline-sql.stage.mozaws.net/queries/116/source#135
36 results - the pie chart is not displayed: https://pipeline-sql.stage.mozaws.net/queries/118/source#139

example 2:
24 results - the pie chart is displayed: https://pipeline-sql.stage.mozaws.net/queries/128/source#159
25 results - the pie chart is not displayed: https://pipeline-sql.stage.mozaws.net/queries/126/source#155

From my tests there is no specific number of results where the pie chart is no longer displayed. It appears this depends on the table the query is executed.

2. When there are long strings displayed in the results there are some display issues. Please view this query: https://pipeline-sql.stage.mozaws.net/queries/123/source#149
and the screenshot: 
![displayissuepiechart](https://user-images.githubusercontent.com/15685441/29971469-ffef5dfa-8f30-11e7-9ac3-8f0283438005.png)

**Comment by alison985** [comment]
Thanks Madalin. #2 is definitely a different and pre-existing bug. Could you please make a new bug for it in Milestone 10?

So far, I can reproduce #1 on staging - the bug shows up in the console. I am pushing a fix for it to staging in PR #244.

**Comment by alison985** [comment]
The 36 pie chart works for me if I start from a new query page and paste the query in: https://pipeline-sql.stage.mozaws.net/queries/130/source#163

**Comment by madalincm** [comment]
I can see the issue but so far I do not have accurate steps to reproduce it.  The only fact that I have so far it is that it depends on the number of results returned by the query. I will mark this issue as verified fixed and open a new one where we can keep investigating it.
Also I will log a new issue for #2
